src/Sparrow/Json/JsonContextPoolBase.cs

```diff
@@ -69,7 +69,7 @@ public bool Add(JsonContextPoolBase<T> parent)
         }
 
         [ThreadStatic]
-        private static ContextStackThreadReleaser _releaser;
+        private ContextStackThreadReleaser _releaser;
```

Comment:
That doesn't actually work. You don't get a separate thread static per value.
An option here is to either use `ThreadLocal` or to have `ContextStackThreadReleaser` hold a list of associated pools

Reply:
The commit message explains it, basically it caused thread id change in one context to affect another one. We suspect it happened because of shared instances of ThreadId members in ContextStackThreadReleaser. After that change above, duplicates in _threadIds were not reproduced.

Reply:
How could that be? The only way cross thread stuff is when this is finalized.
I'm not seeing how that can happen.